docs/architecture/architecture.md

```diff
@@ -0,0 +1,169 @@
+        Container(Backend, "Back End", "FastAPI, Python","Includes the API application, <br/>business logic and system state")
+        ComponentDb(Db, "Database")
+        ContainerQueue(Message queue, "Message queue", "Redis")
+        Component(Workers, "Workers", "Celery")
+
+    }
+
+
+
+Boundary(b2, "External"){
+    System_Ext(Repository, "Repository", "External repository where the <br/> System Card will be stored")
+
+}
+
+UpdateLayoutConfig($c4ShapeInRow="1", $c4BoundaryInRow="4")
+
+Rel(user0, CLI, "Executes tasks locally <br/>from", "Command line")
+UpdateRelStyle(user0, CLI, $offsetY="-40", $offsetX="-140")
+
+Rel(user0, FrontEnd, "Visits Algorithm Management<br/> Toolkit webpage", "HTTPS")
+UpdateRelStyle(user0, FrontEnd, $offsetY="-40", $offsetX="-70")
+
+Rel(CLI, Tasks, "Imports tasks from")
+UpdateRelStyle(CLI, Tasks, $offsetY="-5", $offsetX="-120")
+
+BiRel(FrontEnd, Backend, "Front End makes API calls, <br/> Back End sends live updates", "HTTPS, websocket")
+UpdateRelStyle(FrontEnd, Backend, $offsetY="-10", $offsetX="-165")
+
+
+BiRel(Backend, Db, "Reads form and <br/>writes to", "")
+UpdateRelStyle(Backend, Db, $offsetY="-10", $offsetX="10")
+
+BiRel(Backend, Message queue, "Submits tasks, <br/>gets results")
+UpdateRelStyle(Backend, Message queue, $offsetY="80", $offsetX="15")
+
+
+BiRel(Message queue, Workers, "Accepts tasks, <br/> writes progress")
+UpdateRelStyle(Message queue, Workers, $offsetY="0", $offsetX="10")
+
+
+Rel(Backend, Repository, "Writes System Card to", "")
+UpdateRelStyle(Backend, Repository, $offsetY="20", $offsetX="-40")
+
+Rel(Workers, Tasks, "Imports tasks <br/>from")
+UpdateRelStyle(Workers, Tasks, $offsetY="160", $offsetX="60")
+
+Rel(Tasks, InstrumentRegister, "Imports instructions from", "")
+UpdateRelStyle(Tasks, InstrumentRegister, $offsetY="10", $offsetX="-150")
+
+```
+
+### Walkthrough
+Suppose a user wants to perform a specific task from the Algoritmekader. To execute this task, the user has 2 options.
+
+The first option is to execute the task locally by using the command line interface tool (CLI). The CLI tool imports the task from a Python library named `Tasks`. This library contains executable tasks that implement the measures and instruments from the `Instrument Register` (which are specified in the Algoritmekader). Instructions on how to perform these tasks are imported from the Instrument Register. There exists a one-to-one correspondence between measures and instruments in the `Instrument Register` and the task's within the `Tasks` library.
+
+The second option is to use the Algorithm Management Toolkit (AMT). The user starts by visiting the Algorithm Management Toolkit website. Here, the user encounters a front end interface showing a planning board for projects and tasks. This planning board contains 3 columns: ‘To do’,  ‘Doing’ and ‘Done’. When a user drags a task from ’To do’ to ‘Doing’, the front end makes an API call to the back end of the AMT.
```

Comment:
We haven't really decided on the columns of the kanban board though, it is a great start but also we haven't included the lifecyle of the system/model yet but also maybe we would need a column Blocked or Review. I just want to write it somewhere down here.

Reply:
We could reword it to be more general.

The planning boards contains columns which corresponds to the status of a task, like 'ToDo', 'Doing' and 'Done'. When a user drags a task to re-order tasks or another column, the front-end makes an API call to the back-end of the AMT to update the task accordingly.

Reply:
done